.github/workflows/build_push_img.yml

```diff
@@ -85,32 +124,32 @@ jobs:
 
       - name: Login to Docker Hub
         uses: docker/login-action@v2
-        if: github.event.inputs.push_to == 'DockerHub'
+        if: env.PUSH_TO == 'DockerHub'
         with:
-          username: ${{ github.event.inputs.dockerhub_username }}
-          password: ${{ github.event.inputs.dockerhub_password }}
+          username: ${{ env.DOCKERHUB_USERNAME }}
+          password: ${{ env.DOCKERHUB_PASSWORD }}
 
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v1
-        if: github.event.inputs.push_to == 'ECR'
+        if: env.PUSH_TO == 'ECR'
         with:
           aws-access-key-id: ${{ secrets.ECR_PUSHER_AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.ECR_PUSHER_AWS_SECRET_ACCESS_KEY }}
           aws-region: us-east-1
 
       - name: Login to ECR
         uses: docker/login-action@v2
-        if: github.event.inputs.push_to == 'ECR'
+        if: env.PUSH_TO == 'ECR'
         with:
-          registry: 424082663841.dkr.ecr.us-east-1.amazonaws.com/${{ github.event.inputs.image_repo }}
+          registry: 424082663841.dkr.ecr.us-east-1.amazonaws.com/${{ env.IMAGE_REPO }}
```

Comment:
I think IMAGE_REPO is not required here.

Reply:
The EC2 runner is hosted in us-west-2 and the ECR is in us-east-1. There's cross region traffic when pushing/pulling docker images. Consider using running from the same region.

Reply:
we can specify secrets in `login-action@v3` action which is documented here: https://github.com/marketplace/actions/docker-login#aws-elastic-container-registry-ecr. The `Configure AWS Credentials` is not needed

Reply:
Made this change: https://github.com/marqo-ai/marqo/pull/1024/commits/740097f76bc5fd67568d409c1a680d9ebcb77333#diff-a60798b1e43d7cc0bb54656f01ca73fc84579b6e501acc45d79cbfec3819cb1dR122-R128